### PR TITLE
Core/Player: Update Player::CanFlyInZone to 7.x

### PR DESCRIPTION
**Pull request by Mihc3**
Separated from my Druid Travel Form PR.

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -25665,15 +25665,47 @@ void Player::StoreLootItem(uint8 lootSlot, Loot* loot, AELootResult* aeResult/*
         SendEquipError(msg, nullptr, nullptr, item->itemid);
 }
 
-bool Player::CanFlyInZone(uint32 mapid, uint32 zone) const
+bool Player::CanFlyInZone(uint32 mapid, uint32 areaid) const
 {
     // continent checked in SpellInfo::CheckLocation at cast and area update
-    uint32 v_map = sDB2Manager.GetVirtualMapForMapAndZone(mapid, zone);
+    bool can_fly = false;
+    switch (mapid)
```
Comment (funjoker) [inline src/server/game/Entities/Player/Player.cpp]: where is v_map now ?

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -25665,11 +25665,47 @@ void Player::StoreLootItem(uint8 lootSlot, Loot* loot, AELootResult* aeResult/*
...
 }
 
-bool Player::CanFlyInZone(uint32 mapid, uint32 zone) const
+bool Player::CanFlyInZone(uint32 mapid, uint32 areaid) const
 {
     // continent checked in SpellInfo::CheckLocation at cast and area update
-    uint32 v_map = sDB2Manager.GetVirtualMapForMapAndZone(mapid, zone);
-    return v_map != 571 || HasSpell(54197); // 54197 = Cold Weather Flying
+    bool can_fly = false;
+    switch (mapid)
```
Comment (funjoker) [inline src/server/game/Entities/Player/Player.cpp]: where is v_map now ?

File: src/server/game/Entities/Player/Player.cpp [hunk 3]
```diff
@@ -25665,11 +25665,40 @@ void Player::StoreLootItem(uint8 lootSlot, Loot* loot, AELootResult* aeResult/*
...
+        case 571: // Northrend
+            can_fly = HasSpell(54197); // Cold Weather Flying
+            break;
+        case 870: // Pandaria
+            can_fly = HasSpell(115913); // Wisdom of the Four Winds
+            break;
+        case 1116: // Draenor
+            can_fly = HasSpell(191645); // Draenor Pathfinder
+        case 1220: // Broken Isles
+        case 1704: // Legion Ship - Horizontal - Valsharah
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp]: These are legion invasion scenario maps, you can't fly in them

File: src/server/game/Entities/Player/Player.cpp (hunk 2 above)
Comment (Mihapro) [reply src/server/game/Entities/Player/Player.cpp]: GetVirtualMapForMapAndZone returns strange values, I don't think it's meant for flying, rather for in-game map.
WorldMapArea.db2 - field18 (DisplayMapID)

File: src/server/game/Entities/Player/Player.cpp [hunk 4]
```diff
@@ -25665,11 +25665,40 @@ void Player::StoreLootItem(uint8 lootSlot, Loot* loot, AELootResult* aeResult/*
         SendEquipError(msg, nullptr, nullptr, item->itemid);
 }
 
-bool Player::CanFlyInZone(uint32 mapid, uint32 zone) const
+bool Player::CanFlyInZone(uint32 mapid, uint32 areaid) const
```
Comment (Krudor) [inline src/server/game/Entities/Player/Player.cpp]: If you change the purpose of the parameter, you should rename the method.

File: src/server/game/Entities/Player/Player.cpp (hunk 3 above)
Comment (Mihapro) [reply src/server/game/Entities/Player/Player.cpp]: Ah, I though so. One addon had those ids included 👎

**Comment by Shauren** [comment]
Actually, this entire check is bad and I'm going to remove it

**Pull request closed without merging**